oshi-core/src/main/java/oshi/software/os/mac/MacOperatingSystem.java

```diff
@@ -393,4 +396,92 @@ public NetworkParams getNetworkParams() {
         return new MacNetworkParams();
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public OSService[] getServices() {
+        //Sort by PID
+        OSProcess[] process =  getChildProcesses(1, 0, OperatingSystem.ProcessSort.PID);
+        //Get Directories
+        File ua = new File("~/Library/LaunchAgents");
+        File ga = new File("/Library/LaunchAgents");      
+        File gd = new File("/Library/LaunchDaemons");      
+        File sa = new File("/System/Library/LaunchAgents");      
+        File sd = new File("/System/Library/LaunchDaemons");
+
+        ArrayList<File> files = new ArrayList<File>();
+
+        try {
+            File[] uaFiles = ua.listFiles((dir, name) -> name.toLowerCase().endsWith(".plist"));
+            File[] gaFiles = ga.listFiles((dir, name) -> name.toLowerCase().endsWith(".plist"));
+            File[] gdFiles = gd.listFiles((dir, name) -> name.toLowerCase().endsWith(".plist"));
+            File[] saFiles = sa.listFiles((dir, name) -> name.toLowerCase().endsWith(".plist"));
+            File[] sdFiles = sd.listFiles((dir, name) -> name.toLowerCase().endsWith(".plist"));
+
+            for(File file : uaFiles) files.add(file);
+            for(File file : gaFiles) files.add(file);
+            for(File file : gdFiles) files.add(file);
+            for(File file : saFiles) files.add(file);
+            for(File file : sdFiles) files.add(file);
+
+            OSService[] svcArray = new OSService[files.size()];
+            for (int i = 0; i < svcArray.length; i++) {
+                for (int j = 0; j < process.length; j++) {
+                    svcArray[i].setName(files.get(i).getName()); 
+                    svcArray[i].setState(OSService.State.STOPPED);
+                    ProcTaskAllInfo taskAllInfo = new ProcTaskAllInfo();
+                    if (0 > SystemB.INSTANCE.proc_pidinfo(process[j].getProcessID(), SystemB.PROC_PIDTASKALLINFO, 0, taskAllInfo, taskAllInfo.size())) {
```

Comment:
It also takes a really long time to query services, which is bad.  I think the problem is here, your outer loop has 656 services in it, and you're then looping over this inner loop for every single process.  This ends up O(n^2) with multiple system calls.   Just call the existing `getProcesses()` method once and cache the results in a map with pid as the key and path as the value.... then look up that path here.   If the key exists it's running, otherwise it's stopped.